### DISCUSSION TITLE
Automation

### DISCUSSION THREAD

**Discussion by 4rjunKM** [TrianguloY/URLCheck#406] [Q&A] [answered]
![Screenshot_20241226-221818~2.png](https://github.com/user-attachments/assets/6f2dc8f7-0c78-44e0-bf62-3978d21bc04b)

What is the use of automation settings in the apk

**Comment by TrianguloY** [comment]
You can configure actions based on triggers, like doing a VirusTotal scan to all urls ending with .io, or opening the default app for urls from github.com, etc.

The current options are limited, but I plan to add more triggers and more actions in the future.

**Comment by medo0o0o0** [comment]
> What is the use of automation settings in the apk


https://github.com/user-attachments/assets/21b930fa-c921-4632-be6e-c9adacea1217

**Comment by medo0o0o0** [comment]
Can we automate( HTTP to HTTPS ) and query removal ?

**Reply by TrianguloY** [reply]
the http to https can be automated on the module itself, with `"automatic":true`

the query removal requires a custom pattern, and then it can be automated the same way. (although I wonder if automations could be added for this purpose...)

**Reply by medo0o0o0** [reply]
thank you 🫡

**Reply by medo0o0o0** [reply, marked as answer]
AI helped me wrote it, it works
```
  "Remove query string": {
   "regex": "(\\?[^#]*)?(#.*)?",
   "replacement": "$2",
   "automatic": true
```
